patroni/postgresql/slots.py

```diff
@@ -135,23 +188,46 @@ def __init__(self, postgresql: 'Postgresql') -> None:
         self.schedule()
 
     def _query(self, sql: str, *params: Any) -> Union['cursor', 'Cursor[Any]']:
+        """Helper method for :meth:`PostgreSQL.query`.
+
+        :param sql: SQL statement to execute.
+        :param params: parameters to pass through to :meth:`Postgresql.query`.
+
+        :returns: query response.
+        """
         return self._postgresql.query(sql, *params, retry=False)
 
     @staticmethod
-    def _copy_items(src: Dict[str, Any], dst: Dict[str, Any], keys: Optional[List[str]] = None) -> None:
+    def _copy_items(src: Dict[str, Any], dst: Dict[str, Any], keys: Optional[Collection[str]] = None) -> None:
+        """Select values from *src* dictionary to update in *dst* dictionary for optional supplied *keys*.
+
+        :param src: source dictionary that *keys* will be looked up from.
+        :param dst: destination dictionary to be updated.
+        :param keys: optional list of keys to be looked up in the source dictionary.
+        """
         dst.update({key: src[key] for key in keys or ('datoid', 'catalog_xmin', 'confirmed_flush_lsn')})
```

Comment:
IIRC, in the previous PR you moved default values to default value of the `keys` parameter.

Reply:
Yes, I thought about it a bit and realised that if the arg `keys` was an empty list you'd have no `keys` and therefore no copy. That's different to the current behaviour as there's no test for `is None`. If that's not as intended and you should be able to pass an empty list I'll restore the change.